### ISSUE TITLE
[Command Prompt] : User is not able to Select commands using 'Shift+ right/Left arrow keys' inside Command Prompt.

### ISSUE THREAD

**Issue by Saiteja341** [microsoft/terminal#11985]
### Windows Terminal version

1.12.3472.0

### Windows build number

22509.1011

### Other Software

**Test Environment: **
OS: Windows 11 Version Dev (OS Build 22509.1011)
App: Windows Terminal Preview
Screen Reader: Narrator

### Steps to reproduce

**Repro Steps:** 

1.Open Windows Terminal.
2.Open 'Command prompt' using 'Ctrl + Shift + 2' keys.
3.Now type any text in text-area of 'Command prompt'.
4.Now try to select the command text using 'Shift + right/left arrow'.
5.Observe the issue.

**Users Experience**
Users who rely on keyboard will get impacted here & fail to use the 'Command prompt' effectively, if User is not able to Select commands using 'Shift+ right/Left arrow keys' inside Command Prompt.

**Attachment :** 

[User is not able to Select commands using 'Shift+ right or Left arrow keys' inside Command Prompt..zip](https://github.com/microsoft/terminal/files/7733539/User.is.not.able.to.Select.commands.using.Shift%2B.rightLeft.arrow.keys.inside.Command.Prompt.zip)

### Expected Behavior

User should be able to Select commands using 'Shift+ right/Left arrow keys' inside Command Prompt, which is happening in 'Windows Power shell' tab.

### Actual Behavior

User is not able to Select commands using 'Shift+ right/Left arrow keys' inside Command Prompt.

**Comment by elsaco** [comment]
@Saiteja341 `Shift+RightArrow` is a function of `PSReadLine`. IFIK, `cmd` doesn't use `PSReadLine`.

**Comment by KalleOlaviNiemitalo** [comment]
@elsaco, in PSReadLine, the Shift+arrow combination can only select text in the command that the user is editing. In conhost, the Shift+arrow combination can select the prompt and earlier lines as well.

Similar functionality for Windows Terminal has been requested in <https://github.com/microsoft/terminal/issues/715> so this one could be considered a duplicate of that.

**Comment by elsaco** [comment]
@KalleOlaviNiemitalo you're correct! Also, if the action is started by the mouse pointer, like selecting one char only, then `Shift+RightArrow` or `Shift+LeftArrow` works for extending the selection in a `cmd` tab.

**Comment by zadjii-msft** [comment]
Our plan of record at the moment is to use mark mode (see #5804) to enable this functionality. This won't have exact parity with the vintage console's shift+arrows for selecting text in cmd.exe, but it should be close enough, with the possibility of being more powerful.

**Comment by zadjii-msft** [comment]
This is fixed in Terminal Preview, >=v1.15. The user needs to enter "Mark Mode" (by pressing `ctrl+shift+m`), and then they can select text with `shift+arrows`.

**Comment by ofek** [comment]
Just fyi specifically for shift+left, you'd have to go one character back before highlighting

**Comment by lesleyrs** [comment]
Why does nobody mention WSL? I'd like to be able to do this there too. Is there a reason for why it won't allow highlighting with keys?

**Comment by zadjii-msft** [comment]
Also, adding the following to your settings will work:

```json
  {
      "command":
      {
          "action": "markMode"
      },
      "keys": "shift+left"
  },
```

Sure, it requires an extra keypress, but it'll get the job done.

**Comment by zadjii-msft** [comment]
> It seems there is no way to define actions per profile

That's correct, that's tracked in #5790

**Comment by levicki** [comment]
@zadjii-msft Isn't using standard LUA shortcuts used to work before? I can't even find definition of LUA that doesn't refer to scripting language but I hope you guys working on this project are old enough to know what I mean. Here's a list of shortcuts I'd expect to work in terminal:
```
{
"\UF729"   = "moveToBeginningOfLine:";                       /* Home             */
"@\UF729"  = "moveToBeginningOfDocument:";                   /* Ctrl-Home        */
"$\UF729"  = "moveToBeginningOfLineAndModifySelection:";     /* Shift-Home       */
"@$\UF729" = "moveToBeginningOfDocumentAndModifySelection:"; /* Ctrl-Shift-Home  */
"\UF72B"   = "moveToEndOfLine:";                             /* End              */
"@\UF72B"  = "moveToEndOfDocument:";                         /* Ctrl-End         */
"$\UF72B"  = "moveToEndOfLineAndModifySelection:";           /* Shift-End        */
"@$\UF72B" = "moveToEndOfDocumentAndModifySelection:";       /* Ctrl-Shift-End   */
"\UF72C"   = "pageUp:";                                      /* PageUp           */
"\UF72D"   = "pageDown:";                                    /* PageDown         */
"$\UF728"  = "cut:";                                         /* Shift-Delete     */
"$\UF727"  = "paste:";                                       /* Shift-Insert     */
"@\UF727"  = "copy:";                                        /* Ctrl-Insert      */
"@\UF702"  = "moveWordBackward:";                            /* Ctrl-Left        */
"@\UF703"  = "moveWordForward:";                             /* Ctrl-Right       */
"@$\UF702" = "moveWordBackwardAndModifySelection:";          /* Ctrl-Shift-Left  */
"@$\UF703" = "moveWordForwardAndModifySelection:";           /* Ctrl-Shift-Right */
}
```
This is a list I created when I first started using Mac because I couldn't stand not having those keys work like they did after so many years of using them.

For many of us text selection is core UX. No one wants to “switch modes” just to select text they just entered. Imagine if you had to do that in Notepad every time you wanted to make a selection.

Can you please find some way to allow us to use those keys as intended, at least those that make sense in the context of selecting and editing on the command line?

**Comment by hoffor** [comment]
aaand it's been years with no change. removing fundamental text input functionality and requiring your users to enable a hack that is less efficient is unacceptable imo

i use shift+arrow (and ctrl+shift+arrow which is even less natural with this hack) constantly to highlight text and using my mouse for this absolutely destroys my workflow
